rfd/0106-machine-id-anonymous-telemetry.md

```diff
@@ -0,0 +1,157 @@
+---
+authors: Noah Stride (noah.stride@goteleport.com)
+state: draft
+---
+
+# RFD 0106 - Machine ID Anonymous Agent Telemetry
```

Comment:
I landed here after seeing https://github.com/gravitational/cloud-terraform/pull/1542 fly by.

I don't see any remarks about security. Since this is a public endpoint expecting to accept anonymous traffic, does the design incorporate any sort of rate limiting or validations (outside of the proto schema) to prevent abuse?  grpc strings can be up to 2Gb.

I'm also curious about what the intended behavior is when a security conscious intentionally blocks DNS resolution or traffic for this domain.  Can we make sure this runs in a background goroutine and isn't serialized before anything important.

Reply:
Size limiting will be in place before we actually start using any public endpoint; considering that the service is proxied through cloudflare, can we see if we end up needing rate limiting on our end first?

Reply:
Ah -- cloudflare proxying is a good call. I'm so used to thinking about teleport cloud I forgot we had that in place.  We should do without rate limiting for v1 then -- it isn't going to deliver good ROI to develop that.  Still probably worth explicitly calling out in the design, instead of letting it be implicit.

Reply:
That's not specific to Machine ID, so I'd expect that to be outlined in https://github.com/gravitational/teleport/pull/22400 instead of here.